nixos/modules/services/networking/firewall.nix

```diff
@@ -493,6 +508,22 @@ in
         '';
     };
 
+    networking.firewall.extraAllowed = mkOption {
+      type = types.listOf (types.submodule { options = {
+           source = mkOption { type = types.string; default = ""; example = "example.com"; description = "Source of the packets"; };
+           protocol = mkOption { type = types.enum ["tcp" "udp" "icmp"]; default = ""; example = "tcp"; description = "Protocol used."; };
```

Comment:
What does an empty default do? Note that `""` does not match (or should not match, I didn't test) the valid values in the `enum`. A better type might be `nullOr (listOf (enum ["tcp" "udp" "icmp"]))`, where `null` means any protocol. This also allows specifying multiple protocols (e.g. `["tcp" "udp"]`).